modules/database/src/std/softIoc/softMain.cpp

```diff
@@ -247,8 +248,11 @@ int main(int argc, char *argv[])
             }
 
         } else {
-            if (loadedDb) {
-                epicsThreadExitMain();
```

Comment:
I would much prefer that we change what epicsThreadExitMain() actually does if its current definition makes no sense than just stop using it here, as that would leave the code for the now-unused routine sitting in libCom expecting to be maintained. This is the only call to this routine in Base (there isn't even a test for it), and I'm pretty sure the routine was added for this specific use. Any out-of-tree users would suffer the same "doesn't make sense" problem.

Reply:
If this is truly the only usage found in ~20 years (which I think it is) then wouldn't it make more sense to remove `epicsThreadExitMain()` from libCom?  Since there is no pressing reason to do so, I would first mark it deprecated for a release cycle or two.

Reply:
fyi.  For this sort of "parking" in `main()`.  I see abstracting the signal handling as more useful than the blocking.  cf [pvxs::SigInt](https://mdavidsaver.github.io/pvxs/util.html#_CPPv4N4pvxs6SigIntE).

Reply:
I may be assuming os-specific requirements that don't actually exist, but wouldn't it be better to have one routine (which is what `epicsThreadExitMain()` was supposed to be) that does any cleanup necessary for any application wants to suspend its `main()` thread until the process gets killed from outside?

It ought to be possible to Control-C out of a `softIoc -S` so fixing that would be welcome. I see that `camonitor` calls `ca_pend_event(0)` which we couldn't replace, but `pvget -m` waits on an epicsEvent that (presumably) never gets triggered; the nice thing about using a call to `epicsThreadExitMain()` instead would be that it explicitly documents the author's intention. A portable equivalent of `__attribute__ ((noreturn))` might be a good idea for it, which could also be applied to `cantProceeed()`.

Maybe we could just replace the os-specific versions of `epicsThreadExitMain()` with a generic one that has a looping call to `epicsThreadSuspendSelf()`, as used in `cantProceeed()`.

Reply:
Actually, the unstated assumption may be mine.  I see "suspend until killed" as short-sighted.  It is true that CLIs often start with this, any may never want more.  For these cases, the ~3 line loop+sleep seems utterly trivial.  Why bother with library assistance?

For more complicated cases where cleanup is needed.  imo. the only general way for something like `epicsThreadExitMain()` to handle application cleanup is to return to `main()` and let user code do its thing.  Given that this would break the contract which `epicsThreadExitMain()` has so far made, I don't see that it can reasonably by changed.

> `pvget -m` waits on an epicsEvent that (presumably) never gets triggered

It does, from a signal handler.

fyi. the PVXS version of this (`pvxmonitor`) will also `exit(0)` when all subscriptions "finish".